projects/packages/connection/src/class-manager.php

```diff
@@ -831,11 +831,16 @@ public function connect_user( $user_id = null, $redirect_url = null ) {
 	 * @return Boolean Whether the disconnection of the user was successful.
 	 */
 	public function disconnect_user( $user_id = null, $can_overwrite_primary_user = false, $force_disconnect_locally = false ) {
-		$user_id = empty( $user_id ) ? get_current_user_id() : (int) $user_id;
+		$user_id             = empty( $user_id ) ? get_current_user_id() : (int) $user_id;
+		$connection_owner_id = $this->get_connection_owner_id();
 
 		// Attempt to disconnect the user from WordPress.com.
 		$is_disconnected_from_wpcom = $this->unlink_user_from_wpcom( $user_id );
 
+		if ( $user_id === $connection_owner_id ) {
```

Comment:
I think this should be under `if ( $is_disconnected_from_wpcom || $force_disconnect_locally ) {` as the rest of the user disconnect logic

Reply:
Good catch, thanks!

Reply:
Done https://github.com/Automattic/jetpack/pull/26705/commits/6c97416fd75336adf2215e0d88e6ce54eedba37e